tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Session.java

```diff
@@ -75,7 +76,8 @@
  * <p><b>WARNING:</b>A {@code Session} owns resources that <b>must</b> be explicitly freed by
  * invoking {@link #close()}.
  *
- * <p>Instances of a Session are thread-safe.
+ * <p>Instances of a Session are thread-safe. Modifying the graph from another thread <b>is not
+ * safe</b>.
```

Comment:
I think this should be more explicit. Modifying the graph concurrently while another thread is using a session from that graph is not thread safe.